tests/v2/validation/prime/prime_test.go

```diff
@@ -4,59 +4,91 @@ import (
+	prime "github.com/rancher/rancher/tests/framework/extensions/prime"
 	"github.com/rancher/rancher/tests/framework/extensions/rancherversion"
 	"github.com/rancher/rancher/tests/framework/pkg/config"
 	"github.com/rancher/rancher/tests/framework/pkg/session"
 	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
 	"github.com/stretchr/testify/suite"
 )
 
-const rancherBrand = "suse"
+const (
+	uiBrand         = "ui-brand"
+	defaultRegistry = "system-default-registry"
+	localCluster    = "local"
+)
 
-type PrimeVersionTestSuite struct {
+type PrimeTestSuite struct {
 	suite.Suite
 	session        *session.Session
 	client         *rancher.Client
 	isPrime        bool
 	rancherVersion string
+	primeRegistry  string
 }
 
-func (t *PrimeVersionTestSuite) TearDownSuite() {
+func (t *PrimeTestSuite) TearDownSuite() {
 	t.session.Cleanup()
 }
 
-func (t *PrimeVersionTestSuite) SetupSuite() {
+func (t *PrimeTestSuite) SetupSuite() {
 	testSession := session.NewSession()
 	t.session = testSession
-	primeConfig := new(rancherversion.Config)
 
+	primeConfig := new(rancherversion.Config)
 	config.LoadConfig(rancherversion.ConfigurationFileKey, primeConfig)
+
 	t.isPrime = primeConfig.IsPrime
 	t.rancherVersion = primeConfig.RancherVersion
+	t.primeRegistry = primeConfig.Registry
 
 	client, err := rancher.NewClient("", t.session)
 	assert.NoError(t.T(), err)
+
 	t.client = client
+}
+
+func (t *PrimeTestSuite) TestPrimeUIBrand() {
+	brand, err := t.client.Management.Setting.ByID(uiBrand)
+	if err != nil {
+		t.T().Log(err)
+	}
 
+	err = prime.CheckUIBrand(t.client, t.isPrime, brand)
+	require.NoError(t.T(), err)
 }
 
-func (t *PrimeVersionTestSuite) TestPrimeVersion() {
+func (t *PrimeTestSuite) TestPrimeVersion() {
 	serverConfig, err := rancherversion.RequestRancherVersion(t.client.RancherConfig.Host)
-	assert.Nil(t.T(), err)
+	require.NoError(t.T(), err)
+
+	err = prime.CheckVersion(t.isPrime, t.rancherVersion, serverConfig)
+	require.NoError(t.T(), err)
+}
 
-	assert.Equal(t.T(), t.isPrime, serverConfig.IsPrime)
-	assert.Equal(t.T(), t.rancherVersion, serverConfig.RancherVersion)
-	brand, err := t.client.Management.Setting.ByID("ui-brand")
+func (t *PrimeTestSuite) TestSystemDefaultRegistry() {
+	registry, err := t.client.Management.Setting.ByID(defaultRegistry)
 	if err != nil {
 		t.T().Log(err)
 	}
-	if t.isPrime {
-		assert.Equal(t.T(), rancherBrand, brand.Value)
-	} else {
-		assert.Empty(t.T(), brand.Value)
-	}
+
+	err = prime.CheckSystemDefaultRegistry(t.isPrime, t.primeRegistry, registry)
+	require.NoError(t.T(), err)
+}
+
+func (t *PrimeTestSuite) TestLocalClusterRancherImages() {
+	adminClient, err := rancher.NewClient(t.client.RancherConfig.AdminToken, t.client.Session)
+	require.NoError(t.T(), err)
+
+	clusterID, err := clusters.GetClusterIDByName(adminClient, localCluster)
+	require.NoError(t.T(), err)
+
+	err = prime.CheckLocalClusterRancherImages(t.client, t.isPrime, t.rancherVersion, clusterID)
```

Comment:
if there is any image that isn't using the registry image, we should fail the test

Reply:
Resolved.